src/coins.cpp

```diff
@@ -398,11 +400,19 @@ bool CCoinsViewCache::HavePourRequirements(const CTransaction& tx) const
         }
 
         ZCIncrementalMerkleTree tree;
-        if (!GetAnchorAt(pour.anchor, tree)) {
-            // If we do not have the anchor for the pour,
-            // it is invalid.
+        auto it = intermediates.find(pour.anchor);
+        if (it != intermediates.end()) {
+            tree = it->second;
+        } else if (!GetAnchorAt(pour.anchor, tree)) {
```

Comment:
So this code is checking the local intermediates first for a JoinSplit anchor, and if it doesn't find it then it checks the global tree?